google/cloud/speech/client.py

```diff
@@ -0,0 +1,182 @@
+                            requests. Valid values are: 8000-48000.
+                            16000 is optimal. For best results, set the
+                            sampling rate of the audio source to 16000 Hz.
+                            If that's not possible, use the native sample rate
+                            of the audio source (instead of re-sampling).
+
+        :type language_code: str
+        :param language_code: (Optional) The language of the supplied audio as
+                              BCP-47 language tag. Example: "en-GB".
+                              If omitted, defaults to "en-US".
+
+        :type max_alternatives: int
+        :param max_alternatives: (Optional) Maximum number of recognition
+                                 hypotheses to be returned. The server may
+                                 return fewer than maxAlternatives.
+                                 Valid values are 0-30. A value of 0 or 1
+                                 will return a maximum of 1. Defaults to 1
+
+        :type profanity_filter: bool
+        :param profanity_filter: If True, the server will attempt to filter
+                                 out profanities, replacing all but the
+                                 initial character in each filtered word with
+                                 asterisks, e.g. "f***". If False or omitted,
+                                 profanities won't be filtered out.
+
+        :type speech_context: list
+        :param speech_context: A list of strings (max 50) containing words and
+                               phrases "hints" so that the speech recognition
+                               is more likely to recognize them. This can be
+                               used to improve the accuracy for specific words
+                               and phrases. This can also be used to add new
+                               words to the vocabulary of the recognizer.
+
+        :rtype: list
+        :returns: A list of dictionaries. One dict for each alternative. Each
+                  dictionary typically contains two keys (though not
+                  all will be present in all cases)
+
+                  * ``transcript``: The detected text from the audio recording.
+                  * ``confidence``: The confidence in language detection, float
+                    between 0 and 1.
+        """
+
+        if (content is None) and (source_uri is None):
+            raise ValueError('content and source_uri cannot be both equal to\
+                             None')
+
+        if (content is not None) and (source_uri is not None):
+            raise ValueError('content and source_uri cannot be both different from\
+                             None')
+        if encoding is None:
+            raise ValueError('encoding cannot be None')
+        if sample_rate is None:
+            raise ValueError('sample_rate cannot be None')
+
+        if content is not None:
```

Comment:
OK, I see.  ISTM that both should default to `None`, since we have the checks in place that exactly one is supplied.